dbt/include/snowflake/macros/relations/dynamic_table/create.sql

```diff
@@ -1,16 +1,83 @@
 {% macro snowflake__get_create_dynamic_table_as_sql(relation, sql) -%}
+{#-
+--  Produce DDL that creates a dynamic table
+--
+--  Args:
+--  - relation: Union[SnowflakeRelation, str]
+--      - SnowflakeRelation - required for relation.render()
+--      - str - is already the rendered relation name
+--  - sql: str - the code defining the model
+--  Globals:
+--  - config: NodeConfig - contains the attribution required to produce a SnowflakeDynamicTableConfig
+--  Returns:
+--      A valid DDL statement which will result in a new dynamic table.
+-#}
 
     {%- set dynamic_table = relation.from_config(config.model) -%}
 
+    {%- if dynamic_table.catalog.table_format == 'iceberg' -%}
```

Comment:
Why check this manually through the catalog attribute instead of doing a `is_iceberg_format` method call on the relation?

Reply:
One of the goals of using `RelationConfig` is to reduce the dependency on the global variables, e.g. `relation` and `config`. If we were able to not rely on any globals, then we could test an individual macro without needing to create any jinja context. So while using `relation.is_iceberg_format` is the right approach on other relation types, I feel like this aligns with the `RelationConfig` approach.

Reply:
Makes sense. I like the idea of catalog becoming a first class citizen for dispatching code paths, especially after we talked about this as a team.